lib/bolt/inventory/inventory2.rb

```diff
@@ -110,13 +110,13 @@ def features(target)
 
       def data_hash
         {
-          data: @data,
+          data: {},
           target_hash: {
-            target_vars: @target_vars,
-            target_facts: @target_facts,
```

Comment:
This should ensure that get_targets always returns an empty array. Previously the catalog data was serializing the object instance of `Concurrent::Delay` in a way that would not be usable within the apply block. https://github.com/puppetlabs/bolt/blob/71d37234dc8c4635c147c5fdd6c650649cf7dec7/lib/bolt/applicator.rb#L104
Example plan:
```
plan test(TargetSpec $nodes){
  apply($nodes){
  	notice(get_targets('all'))
  }
}
```
Inventory v1
```yaml
nodes:
  - name: s0b58ex4u43wq6y.delivery.puppetlabs.net
    alias: linux
    config:
      transport: ssh
      ssh:
        host-key-check: false
        user: root
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt plan run test -t linux
Starting: plan test
Starting: plan test
Starting: apply catalog on s0b58ex4u43wq6y.delivery.puppetlabs.net
s0b58ex4u43wq6y.delivery.puppetlabs.net: Scope(Class[main]): [Target('s0b58ex4u43wq6y.delivery.puppetlabs.net', {"user"=>"root"})]
Finished: apply catalog with 0 failures in 4.41 sec
Finished: plan test in 4.41 sec
Finished: plan test in 4.41 sec
Plan completed successfully with no result
```
inventory v2
```yaml
version: 2
targets: 
  - uri: s0b58ex4u43wq6y.delivery.puppetlabs.net
    alias: linux
    config:
      transport: ssh
      ssh:
        host-key-check: false
        user: root
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt plan run test -t linux
Starting: plan test
Starting: plan test
Starting: apply catalog on s0b58ex4u43wq6y.delivery.puppetlabs.net
s0b58ex4u43wq6y.delivery.puppetlabs.net: Scope(Class[main]): []
Finished: apply catalog with 0 failures in 4.4 sec
Finished: plan test in 4.41 sec
Finished: plan test in 4.41 sec
Plan completed successfully with no result
```